context/src/test/java/io/grpc/ContextTest.java

```diff
@@ -759,7 +759,8 @@ public void initContextWithCustomClassLoaderWithCustomLogger() throws Exception
     StaticTestingClassLoader classLoader =
         new StaticTestingClassLoader(
             getClass().getClassLoader(),
-            Pattern.compile("(io\\.grpc\\.Context.*)|(io\\.grpc\\.ThreadLocalContextStorage.*)"));
+            Pattern.compile("(io\\.grpc\\.Context.*)|(io\\.grpc\\.ThreadLocalContextStorage.*)|"
+                + "(io\\.grpc\\.PersistentHashArrayMappedTrie)"));
```

Comment:
Oh, the package visibility kicks in because... I guess we are implicitly redefining the package. And it doesn't break further because you aren't adding any keys to the context. For sanity's sake, add `.*` after `Trie` or replace the entire string with `"io\\.grpc\\.[^.]*"` instead.